uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/Workbench.java

```diff
@@ -248,20 +265,10 @@ private void bootstrap() {
 
     private void addCloseHandler() {
         if (UserAgent.isChrome()) {
-            Window.addCloseHandler(event -> workbenchCloseHandler.onWindowClose(workbenchCloseCommand));
-
-            Window.addWindowClosingHandler(event -> {
-                if (!placeManager.canCloseAllPlaces()) {
-                    // Setting a non-null message will make the browser to present a confirmation dialog that asks the user
-                    // whether or not they wish to navigate away from the page. The message in the dialog, however,
-                    // is not customizable anymore mainly due to scammers using such a message to trick people.
-                    // Thus, each browser shows its own message. If the user has an outdated browser, then our custom
-                    // message might be shown.
-                    event.setMessage(WorkbenchConstants.INSTANCE.closingWindowMessage());
-                }
-            });
+            Window.addCloseHandler(event -> workbenchCloseHandler.onWindowClose(workbenchCloseCommand, event));
+            Window.addWindowClosingHandler(event -> workbenchCloseHandler.onWindowClosing(workbenchClosingCommand, event));
         } else {
-            Window.addWindowClosingHandler(event -> workbenchCloseHandler.onWindowClose(workbenchCloseCommand));
+            Window.addWindowClosingHandler(event -> workbenchCloseHandler.onWindowClose(workbenchCloseCommand, null));
```

Comment:
@manstis Thanks for the review. I've made a few changes that should address your comments.

About your code suggestion, this can't be done this way because, for other browsers than Chrome, we need to register the `workbenchCloseHandler.onWindow**Close**` in the Window.addWindow**Closing**Handler, because the `Window.addCloseHandler` does not work as expected on them. I've also added a comment in the code about that now.